### ISSUE TITLE
High query cout on user

### ISSUE THREAD

**Issue by mrfoxes** [flavors/django-graphql-jwt#164]
I'm on Django 3 and i'm noticing that a high number of query happen at some point.

Have anyone experienced the same behaviour?

<img width="1152" alt="Screenshot 2020-01-29 at 12 03 06" src="https://user-images.githubusercontent.com/17138445/73351924-4cd56a00-4290-11ea-86f5-44c393dd5d04.png">

**Comment by TitanFighter** [comment]
You may not get any response, because not many people migrated to Django 3 :)

**Comment by JSeam2** [comment]
I haven't really looked into this, but this may be possibly due to ASGI being introduced in Django 3.0

**Comment by mongkok** [comment]
Hey @MrFoxes ,
I'm working with Django 3 and I haven't noticed this behavior.
Could you detail about your JWT settings and the GraphQL query executed?

**Comment by MrFoxes** [comment]
> Hey @MrFoxes ,
> I'm working with Django 3 and I haven't noticed this behavior.
> Could you detail about your JWT settings and the GraphQL query executed?

@mongkok I'm noticing this behaviour because the middleware run for every resolver function so if i decide to query a lot of fields every time i get the AuthMiddleware running. It may not be an issue related with Django 3 but with graphene maybe.

As you can see also at this issue we were discussing about this behaviour and there are some interesting points

graphql-python/graphene-django#810

Also have a look at what's inside [graphql-utils](https://pypi.org/project/graphql-utilities/0.1.0/)

For the moment i was doing something like this modifying the AuthMiddleware a little bit but i haven't tried other solutions yet.

Adding `root in None` prevent the heavy logic re-run but don't prevent the middleware itself from running for every resolver

`
def resolve(self, next, root, info, **kwargs):
        if root is None:
            context = info.context
            token_argument = get_token_argument(context, **kwargs)

            if jwt_settings.JWT_ALLOW_ARGUMENT and token_argument is None:
                user = self.cached_authentication.parent(info.path)
`

**Comment by mongkok** [comment]
Hey @MrFoxes ,
Yes, Graphene's middleware runs for every resolver, I don't think it's an issue if you know how to handle.

Is `JWT_ALLOW_ARGUMENT` True in your `GRAPHQL_JWT` settings?
Is the  `SessionMiddleware` included in your `settings.MIDDLEWARE`?

**Comment by MrFoxes** [comment]
@mongkok Yes JWT_ALLOW_ARGUMENT is set to True and all middlewares are properly configured

**Comment by mongkok** [comment]
@MrFoxes I need more information:
- What is the GraphQL query? You can rename the field names to a,b,c...
- What cookies are sent? sessionid? JWT?
- Authorization header?

**Comment by r-0-0-t** [comment]
The culprit is **JWT_ALLOW_ARGUMENT** , set it to FALSE
